From 04447f7aab27eccb900cf9a63169fd50d0f895eb Mon Sep 17 00:00:00 2001
From: Ioana Radulescu <ruxandra.radulescu@nxp.com>
Date: Tue, 18 Dec 2018 16:35:00 +0200
Subject: [PATCH 105/741] dpaa2-eth: Increase Rx FQ taildrop threshold

commit 3bb05b45c25b59a5851a5c80ec10bc67ab66f236 from
http://source.codeaurora.org/external/qoriq/qoriq-components/linux

Bump up to 1MB the taildrop threshold for each Rx frame queue.
This ensures a reasonable number of frames get buffered on ingress
even for larger frame sizes, making the system more resilient to
traffic bursts or spikes in cpu load.

Signed-off-by: Ioana Radulescu <ruxandra.radulescu@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 drivers/net/ethernet/freescale/dpaa2/dpaa2-eth.h | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/net/ethernet/freescale/dpaa2/dpaa2-eth.h b/drivers/net/ethernet/freescale/dpaa2/dpaa2-eth.h
index 3e3d2a690296..ae3b2bed0545 100644
--- a/drivers/net/ethernet/freescale/dpaa2/dpaa2-eth.h
+++ b/drivers/net/ethernet/freescale/dpaa2/dpaa2-eth.h
@@ -39,11 +39,11 @@
 /* Maximum burst size value for Tx shaping */
 #define DPAA2_ETH_MAX_BURST_SIZE	0xF7FF
 
-/* Set the taildrop threshold (in bytes) to allow the enqueue of several jumbo
- * frames in the Rx queues (length of the current frame is not
- * taken into account when making the taildrop decision)
+/* Set the taildrop threshold to 1MB to allow the enqueue of a sufficiently
+ * large number of jumbo frames in the Rx queues (length of the current frame
+ * is not taken into account when making the taildrop decision)
  */
-#define DPAA2_ETH_TAILDROP_THRESH	(64 * 1024)
+#define DPAA2_ETH_TAILDROP_THRESH	(1024 * 1024)
 
 /* Maximum number of Tx confirmation frames to be processed
  * in a single NAPI call
-- 
2.17.1

